hashes/zkevm/src/keccak/coprocessor/circuit/leaf.rs

```diff
@@ -365,17 +382,16 @@ impl<F: Field> KeccakCoprocessorLeafCircuit<F> {
     }
 
     /// Publish outputs of the circuit as public instances.
-    fn publish_outputs(
-        &self,
-        gate: &impl GateInstructions<F>,
-        outputs: &[KeccakCircuitOutput<AssignedValue<F>>],
-    ) {
+    fn publish_outputs(&self, outputs: &[KeccakCircuitOutput<AssignedValue<F>>]) {
+        // The length of outputs should always equal to params.capacity.
+        assert_eq!(outputs.len(), self.params.capacity);
         if !self.params.publish_raw_outputs {
+            let range_chip = self.base_circuit_builder.borrow().range_chip();
```

Comment:
I made the `gate` part of function input because this creates a new `range_chip`. It's mostly smart pointers, but it seemed better to just re-used the `range_chip` already constructed from earlier in `synthesize`?